### PR TITLE
Provide some missing method declarations for posixlib stdlib.scala

### PR DESCRIPTION
**Pull request by LeeTibbert**
This PR provides some methods previously missing in posixlib stdlib.scala.

It is a reduction-in-strength partial fix to Issue #3650.  That Issue should remain open so that type declarations
missing from this PR eventually get added.

This PR removes impediments to fixing Issues #3690 and, eventually,  #3651.

###### Backporting: 
Because this PR moves/corrects the `realpath()` location from clib to posixlib, it is probably not an economic candidate
for backporting.

### CODE REVIEW

File: javalib/src/main/scala/java/io/File.scala [hunk 1]
```diff
@@ -13,6 +13,7 @@ import scala.scalanative.libc.stdio._
 import scala.scalanative.libc.stdlib._
 import scala.scalanative.libc.string._
 import scala.scalanative.nio.fs.FileHelpers
+import scala.scalanative.posix
```
Comment (ekrich) [inline javalib/src/main/scala/java/io/File.scala:16]:
```suggestion
import scala.scalanative.posix.{stdlib => pstdlib}
```

File: javalib/src/main/scala/java/io/File.scala [hunk 2]
```diff
@@ -299,7 +300,7 @@ class File(_path: String) extends Serializable with Comparable[File] {
       fromCWideString(resolvedName, StandardCharsets.UTF_16LE)
     } else {
       val resolvedName: Ptr[Byte] = alloc[Byte](limits.PATH_MAX)
-      if (realpath(toCString(path), resolvedName) == null) {
+      if (posix.stdlib.realpath(toCString(path), resolvedName) == null) {
```
Comment (ekrich) [inline javalib/src/main/scala/java/io/File.scala:303]:
```suggestion
      if (pstdlib.realpath(toCString(path), resolvedName) == null) {
```

Or you might not need the prefix if you moved `realpath`.

**Review by ekrich — COMMENTED** [review]
You might want to experiment with the imports.

File: javalib/src/main/scala/java/io/File.scala (hunk 2 above)
Comment (LeeTibbert) [reply javalib/src/main/scala/java/io/File.scala]: Eric,

Thank you for the timely review & suggestions.

You encouraged me to do a simpler fix in File.scala, based on the work that you and Wojchiech
had done to layer many files, including stdlib, in posixlib over the ISO C files in clib.  Might as
well benefit from that work.

Anybody in File.scala to begin with is in a world of hurtin'. They need all the simplicity & 
transparency they can get.

**Comment by LeeTibbert** [comment]
I believe this PR is ready for final review.

There are a number of CI failures which appear to have nothing to do with this PR.
Let's see what success the re-try pixies may bring.

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**